pandas/core/frame.py

```diff
@@ -819,13 +819,20 @@ def iteritems(self):
         koala    80000
         Name: population, dtype: int64
         """
+
+    @Appender(_shared_docs["items"] % "Yields\n        ------")
+    def items(self):
         if self.columns.is_unique and hasattr(self, "_item_cache"):
             for k in self.columns:
                 yield k, self._get_item_cache(k)
         else:
             for i, k in enumerate(self.columns):
                 yield k, self._ixs(i, axis=1)
 
+    @Appender(_shared_docs["items"] % "Returns\n        -------")
```

Comment:
Both return a generator, so why the different docstring here?

Reply:
It's the "return " part of the method that makes flake8 think this is not a generator, so it won't allow a "Yields" section in the iteritems doc string.

I think flake8 is tecnically right, but it makes for uglier code in this case.

Reply:
On second thought, I think maybe if the Appender is followed by a ``# noqa`` comment, we can make flake8 ignore that section. I'll try it tomorrow.